test/RelayServer.test.ts

```diff
@@ -1109,43 +1111,72 @@ contract('RelayServer', function (accounts) {
-        let receipts = await newServer._registerIfNeeded()
-        assertRelayAdded(receipts, newServer)
-        receipts = await newServer._registerIfNeeded()
-        assert.equal(receipts.length, 0, 'should not re-register if already registered')
+
+      function registrationTests (): void {
+        it('register server and add workers when not registered', async function () {
+          const receipts = await newServer._registerIfNeeded()
+          assertRelayAdded(receipts, newServer)
+        })
+        it('do not register server when already registered', async function () {
+          let receipts = await newServer._registerIfNeeded()
+          assertRelayAdded(receipts, newServer)
+          receipts = await newServer._registerIfNeeded()
+          assert.equal(receipts.length, 0, 'should not re-register if already registered')
+        })
+        it('re-register server when params changed', async function () {
+          let receipts = await newServer._registerIfNeeded()
+          assertRelayAdded(receipts, newServer)
+          // @ts-ignore
+          newServer.baseRelayFee++
+          receipts = await newServer._registerIfNeeded()
+          assertRelayAdded(receipts, newServer, false)
+          // @ts-ignore
+          newServer.pctRelayFee++
+          receipts = await newServer._registerIfNeeded()
+          assertRelayAdded(receipts, newServer, false)
+          // @ts-ignore
+          newServer.url = 'fakeUrl'
+          receipts = await newServer._registerIfNeeded()
+          assertRelayAdded(receipts, newServer, false)
+        })
+      }
+
+      describe('without re-registration', function () {
+        beforeEach(async function () {
+          id = (await snapshot()).result
+          newServer = await bringUpNewRelay()
+          // @ts-ignore
+          newServer.authorizedHub = true
+          const stake = await newServer.refreshStake()
+          assert.deepEqual(stake, oneEther)
+          assert.equal(newServer.owner, relayOwner, 'owner should be set after refreshing stake')
+          assert.equal(newServer.registrationBlockRate, undefined)
+        })
+        afterEach(async function () {
+          await revert(id)
+        })
+        registrationTests()
       })
-      it('re-register server when params changed', async function () {
-        let receipts = await newServer._registerIfNeeded()
-        assertRelayAdded(receipts, newServer)
-        // @ts-ignore
-        newServer.baseRelayFee++
-        receipts = await newServer._registerIfNeeded()
-        assertRelayAdded(receipts, newServer, false)
-        // @ts-ignore
-        newServer.pctRelayFee++
-        receipts = await newServer._registerIfNeeded()
-        assertRelayAdded(receipts, newServer, false)
-        // @ts-ignore
-        newServer.url = 'fakeUrl'
-        receipts = await newServer._registerIfNeeded()
-        assertRelayAdded(receipts, newServer, false)
+      describe('with re-registration', function () {
```

Comment:
I'm trying to understand why you re-run all tests again, where the only test that can be affected is the added one.
why not just modify its configuration to see that (say) after 900 blocks i doens't re-register, and after 1100 it does ?

Reply:
I don't understand the question but I don't see an answer either 😀

Reply:
Fixed